server/subsonic/stream.go

```diff
@@ -106,24 +127,49 @@ func (c *StreamController) Download(w http.ResponseWriter, r *http.Request) (*re
 
 	switch v := entity.(type) {
 	case *model.MediaFile:
-		stream, err := c.streamer.NewStream(ctx, id, "raw", 0)
+		stream, err := c.streamer.NewStream(ctx, id, format, maxBitRate)
+
 		if err != nil {
 			return nil, err
 		}
 
 		disposition := fmt.Sprintf("attachment; filename=\"%s\"", stream.Name())
 		w.Header().Set("Content-Disposition", disposition)
-		http.ServeContent(w, r, stream.Name(), stream.ModTime(), stream)
+
+		if stream.Seekable() {
+			http.ServeContent(w, r, stream.Name(), stream.ModTime(), stream)
+		} else {
+			w.Header().Set("Accept-Ranges", "none")
+			w.Header().Set("Content-Type", stream.ContentType())
+
+			estimateContentLength := utils.ParamBool(r, "estimateContentLength", false)
+
+			if estimateContentLength {
+				length := strconv.Itoa(stream.EstimatedContentLength())
+				log.Trace(ctx, "Estimated content-length", "contentLength", length)
+				w.Header().Set("Content-Length", length)
+			}
+
+			if r.Method == "HEAD" {
+				go func() { _, _ = io.Copy(io.Discard, stream) }()
+			} else {
+				if c, err := io.Copy(w, stream); err != nil {
+					log.Error(ctx, "Error sending transcoded file", "id", id, err)
+				} else {
+					log.Trace(ctx, "Success sending transcode file", "id", id, "size", c)
+				}
+			}
+		}
```

Comment:
This is basically duplicated from the `Stream()` method above. Can we extract this into its own method?